### ISSUE TITLE
static-property-placement reports violations on non-React classes

### ISSUE THREAD

**Issue by jp7837** [yannickcr/eslint-plugin-react#2884]
Version 7.21.5

A `static-property-placement` violation is reported on a non-React class e.g.

```typescript
class Foo {
    static context: {};
}
```

Should it not check that the class extends `React.Component`?

**Comment by ljharb** [comment]
Indeed it should - the rule is using our component detection, so I'm not sure why it's warning here.
